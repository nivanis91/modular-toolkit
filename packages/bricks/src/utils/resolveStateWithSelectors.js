import { makeWorkWithGlobalState } from '@modular-toolkit/selectors';

function resolveStateWithSelectors(store, state) {
    return Object.entries(state).reduce((acc, [key, value]) => {
        switch (typeof value) {
            case 'function':
                return { ...acc, [key]: makeWorkWithGlobalState(value)(store.getState()) };
            case 'object':
                return { ...acc, [key]: resolveStateWithSelectors(store, value) };
            default:
                return { ...acc, [key]: value };
        }
    }, {});
}

export default resolveStateWithSelectors;
