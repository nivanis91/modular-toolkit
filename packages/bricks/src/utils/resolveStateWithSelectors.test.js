import resolveStateWithSelectors from './resolveStateWithSelectors';

const globalState = {
    bar: {
        baz: {
            qux: {
                quux: 'QUUX'
            }
        }
    }
};

const store = {
    getState() {
        return globalState;
    }
};

const selector = s => s.bar.baz.qux;

jest.mock('@modular-toolkit/selectors', () => ({
    makeWorkWithGlobalState: s => s
}));

const localState = {
    foo: 'FOO',
    fasel: {
        bar: selector
    }
};

test('The result of calling resolveStateWithSelectors is correct', () =>
    expect(resolveStateWithSelectors(store, localState)).toMatchSnapshot());
