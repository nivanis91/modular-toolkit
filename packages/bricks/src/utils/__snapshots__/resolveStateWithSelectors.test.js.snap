// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The result of calling resolveStateWithSelectors is correct 1`] = `
Object {
  "fasel": Object {
    "bar": Object {
      "quux": "QUUX",
    },
  },
  "foo": "FOO",
}
`;
